cli/args/lockfile.rs

```diff
@@ -91,8 +92,9 @@ impl CliLockfile {
     // do an atomic write to reduce the chance of multiple deno
     // processes corrupting the file
     atomic_write_file_with_retries(
+      &FsSysTraitsAdapter::new_real(),
```

Comment:
This `FsSysTraitsAdapter` is an adapter for the traits in `sys_traits` and `deno_fs::FileSystem`. It's temporary though. After this PR, I'll decouple the CLI crate's code from `deno_fs` as that makes extracting the code from the CLI crate a lot easier.